### PR TITLE
unix: close signal pipe fds when last loop is done

### PR DESCRIPTION
**Pull request by bnoordhuis**
Fixes #875.

Adds a leaked file descriptor check to the test runner that scans for open file descriptors when the test is done.  Most of the other commits are fix-ups that plug fd leaks in the tests.  I just noticed that the commit log that adds the check is a bit terse, I'll fix that up before landing this.

@indutny or @saghul, feel free to review it if you like.

### CODE REVIEW

File: test/test-tcp-close.c [hunk 1]
```diff
@@ -131,6 +131,9 @@ static void start_server(uv_loop_t* loop, uv_tcp_t* handle) {
   ASSERT(write_cb_called == NUM_WRITE_REQS);
   ASSERT(close_cb_called == 1);
 
+  uv_close((uv_handle_t*) &tcp_server, NULL);
+  ASSERT(0 == uv_run(loop, UV_RUN_NOWAIT));
```
Comment (saghul) [inline test/test-tcp-close.c]: Any specific reason to use noawit here instead of default like in the other cases?

**Comment by saghul** [comment]
LGTM. I guess those test failures are just spurious, right?

File: test/test-tcp-close.c (hunk 1 above)
Comment (bnoordhuis) [inline test/test-tcp-close.c]: Hum, that's probably auto-complete fail.  Thanks for pointing that out.

**Comment by bnoordhuis** [comment]
> I guess those test failures are just spurious, right?

They're probably legitimate insofar that I suspect that Jenkins is doing something to the test environment.  Guess I'll have to dig into that.

**Comment by indutny** [comment]
Few nits, but generally LGTM.

**Comment by indutny** [comment]
@bnoordhuis sorry for a delay, may I ask you to rebase it off the current master? I tried doing it, but got 17 test failures on osx...

**Comment by bnoordhuis** [comment]
I don't really have an interest in libuv anymore.  I'll close the PR but feel free to pick up where I left off.

**Comment by indutny** [comment]
@bnoordhuis sure, though, it is a bit sad, you know.

**Pull request closed without merging**